src/fireEvent.js

```diff
@@ -6,30 +6,48 @@ const findEventHandler = (
   element: ReactTestInstance,
   eventName: string,
   callsite?: any,
-  nearestHostDescendent?: ReactTestInstance
+  nearestHostDescendent?: ReactTestInstance,
+  hasDescendandHandler?: boolean
 ) => {
+  const handler = getEventHandler(element, eventName);
+  const hasHandler = handler != null || hasDescendandHandler;
+
   const isHostComponent = typeof element.type === 'string';
   const hostElement = isHostComponent ? element : nearestHostDescendent;
   const isEventEnabled =
     hostElement?.props.onStartShouldSetResponder?.() !== false;
+  if (handler && isEventEnabled) return handler;
 
-  const eventHandlerName = toEventHandlerName(eventName);
+  // Do not bubble event to the root element
+  if (element.parent === null || element.parent.parent === null) {
+    if (hasHandler) return null;
+    else
```

Comment:
please use curly braces in if/else statements. I find it easier to scan through the code, especially if expressions inside are multiline